tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_NormalCovectorAndMagnitude.cpp

```diff
@@ -0,0 +1,298 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <optional>
+#include <unordered_map>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/Identity.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.tpp"
+#include "Domain/FaceNormal.hpp"
+#include "Domain/InterfaceComputeTags.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DiscontinuousGalerkin/Actions/NormalCovectorAndMagnitude.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+using Affine = domain::CoordinateMaps::Affine;
+using Affine2D = domain::CoordinateMaps::ProductOf2Maps<Affine, Affine>;
+using Affine3D = domain::CoordinateMaps::ProductOf3Maps<Affine, Affine, Affine>;
+
+template <size_t VolumeDim>
+auto make_affine_map() noexcept;
+
+template <>
+auto make_affine_map<1>() noexcept {
+  return domain::make_coordinate_map_base<Frame::Grid, Frame::Inertial>(
+      Affine{-1.0, 1.0, -0.3, 0.7});
+}
+
+template <>
+auto make_affine_map<2>() noexcept {
+  return domain::make_coordinate_map_base<Frame::Grid, Frame::Inertial>(
+      Affine2D{Affine{-1.0, 1.0, -0.3, 0.7}, Affine{-1.0, 1.0, 0.3, 0.55}});
+}
+
+template <>
+auto make_affine_map<3>() noexcept {
+  return domain::make_coordinate_map_base<Frame::Grid, Frame::Inertial>(
+      Affine3D{Affine{-1.0, 1.0, -0.3, 0.7}, Affine{-1.0, 1.0, 0.3, 0.55},
+               Affine{-1.0, 1.0, 2.3, 2.8}});
+}
+
+struct ScaleZeroIndex : db::SimpleTag {
+  using type = double;
+};
+
+template <size_t Dim>
+struct SimpleUnnormalizedFaceNormal
+    : db::ComputeTag,
+      domain::Tags::UnnormalizedFaceNormal<Dim> {
+  using base = domain::Tags::UnnormalizedFaceNormal<Dim>;
+  using return_type = typename base::type;
+  static void function(const gsl::not_null<return_type*> result,
+                       const double scale_zero_index,
+                       const Mesh<Dim - 1>& face_mesh,
+                       const Direction<Dim>& direction) noexcept {
+    for (size_t i = 0; i < Dim; ++i) {
+      result->get(i) =
+          DataVector{face_mesh.number_of_grid_points(),
+                     i == direction.dimension() ? 1.0 * i + 0.25 : 0.0};
+    }
+    result->get(0) *= scale_zero_index;
+  }
+  using argument_tags = tmpl::list<ScaleZeroIndex, domain::Tags::Mesh<Dim - 1>,
+                                   domain::Tags::Direction<Dim>>;
+  using volume_tags = tmpl::list<ScaleZeroIndex>;
+};
+
+struct FlatSpaceSystem {};
+
+template <size_t Dim>
+struct InverseSpatialMetric : db::SimpleTag {
```

Comment:
why not use the inverse spatial metric tag from the gr namespace?

Reply:
I don't want `Evolution/DG` (even the tests) depending on GR.